src/chainrules.jl

```diff
@@ -0,0 +1,15 @@
+using ChainRulesCore
+
+_keyless_unname_pullback(ȳ) =  (NoTangent(), ȳ)
+_keyless_unname_pullback(ȳ::Tangent) = __keyless_unname_pullback(ȳ.data)
+_keyless_unname_pullback(ȳ::AbstractThunk) = _keyless_unname_pullback(unthunk(ȳ))
+
+function ChainRulesCore.rrule(::typeof(keyless_unname), x::KeyedArray)
+    project_x = ProjectTo(x.data)
```

Comment:
I think we should actually define a `ProjectTo(x::KeyedArray)` that returns a `ProjectTo{KeyedArray}` which not only projects the backing data, but also reattaches the keys from the primal to the tangent.